README.md

```diff
@@ -137,6 +156,67 @@ Smart contract interaction for the following scenarios:
+
+Name | Data Type | Description
+--- | --- | ---
+giveQuantity | `uint` | Quantity of Melon tokens to invest
+shareQuantity | `uint` | Quantity of fund shares to receive
+incentiveQuantity | `uint` | Quantity of Melon tokens to award the entity executing the request
+
+2. R parameters are checked against restriction rules specified in the participation module P by the boolean function P.isSubscriptionPermitted (E.g Participant being an attested Uport identity).
+3. R is then executed in by any entity via F.executeRequest after certain conditions are satisifed.  These conditions include if *currentTimestamp - R.timestamp >= DF.INTERVAL* (DF refers to Datafeed module and INTERVAL corresponds to update frequency value) and if *DF.getLastUpdateId >= R.lastDataFeedUpdateId + 2*. This is to minimize unfair advantage from information asymmetries associated with the investor.
+
+**Participant redeems from a Melon fund**
+
+1. A participant can redeem from a fund by first creating a redemption request. Parameters to be specified are:
+
+Name | Data Type | Description
+--- | --- | ---
+shareQuantity | `uint` | Quantity of fund shares to redeem
+receiveQuantity | `uint` | Quantity of Melon tokens to receive in return
+incentiveQuantity | `uint` | Quantity of Melon tokens to award the entity executing the request
+
+2. Request parameters are checked against restriction rules specified in the participation module P via the boolean function P.isRedemptionPermitted.
+3. R is then executed in the exact same way as mentioned earlier.
+
+**Manager makes an order**
+
+1. Manager can make an order by specifying asset pair, sell and buy quantities as parameters. Order parameters are checked against restriction rules specified in the risk management module R via the boolean function R.isMakePermitted.
+2. Order is then placed on the selected exchange through the exchangeAdapter contract E via E.makeOrder by specifying the exchange and order parameters as parameters.
+3. The order is filled on the selected exchange (In future, can be any compatible decentralized exchange like OasisDex, Kyber, e.t.c)  when the price is met.
+
+**Manager takes an orders**
+
+1. Manager can take an order by specifying an order id and quantity as parameters. Order parameters are checked against restriction rules specified in the risk management module R via the boolean function R.isTakePermitted.
+2. Order id must correspond to a valid, existing order on the selected exchange. Order is then placed on the selected exchange through the exchangeAdapter contract E via E.takeOrder by specifying the exchange and order parameters as parameters.
+
+**Manager converts rewards into shares**
+
+1. Manager rewards in the form of ownerless shares of the fund F can be allocated to the manager via F.convertUnclaimedRewards function. First internal stats of F are calculated using F.performCalculations function.
```

Comment:
here a sentence describing ownerless shares would be good.

eg quantity of shares, representing unclaimed rewards by the Manager, such as rewards for managing the fund and for performance.